### PR TITLE
Setup Android side of MLVision text and created example app

### PR DESCRIPTION
**Pull request by bparrishMines**

### CODE REVIEW

File: packages/firebase_ml_vision/example/lib/main.dart [hunk 1]
```diff
@@ -2,32 +2,151 @@
...
+    final Image image = Image.file(_imageFile);
+    final Size imageSize = await _getImageSize(image);
+
+    setState(() {
+      _textLocations = blocks;
+      _imageSize = imageSize;
+    });
+  }
+
+  Future<Size> _getImageSize(Image image) {
```
Comment (kroikie) [inline packages/firebase_ml_vision/example/lib/main.dart]: Nit: Maybe this should be _getImageDimenisons

File: packages/firebase_ml_vision/example/lib/main.dart [hunk 2]
```diff
@@ -2,32 +2,151 @@
...
+        canvas.drawRect(scaleRect(line), paint);
+      }
+
+      paint.color = Colors.red;
+      canvas.drawRect(scaleRect(block), paint);
+    }
+  }
+
+  @override
+  bool shouldRepaint(CustomPainter oldDelegate) => false;
```
Comment (kroikie) [inline packages/firebase_ml_vision/example/lib/main.dart]: Why is this always false, shouldn't it repaint if the image or the identified text changes?

File: packages/firebase_ml_vision/example/lib/main.dart (hunk 1 above)
Comment (bparrishMines) [reply packages/firebase_ml_vision/example/lib/main.dart]: Normally I prefer the word dimensions over size, but the method here actually returns a Size object. I think it makes more sense to use size in this context with this being the case.

File: packages/firebase_ml_vision/android/src/main/java/io/flutter/plugins/firebasemlvision/FirebaseMlVisionPlugin.java [hunk 3]
```diff
@@ -1,26 +1,123 @@
...
   }
+
+  private void handleTextDetectionResult(MethodCall call, final Result result) {
+    File file = new File((String) call.arguments);
+
+    FirebaseVisionImage image;
+    try {
+      image = FirebaseVisionImage.fromFilePath(registrar.context(), Uri.fromFile(file));
+    } catch (IOException exception) {
+      exception.printStackTrace();
```
Comment (kroikie) [inline packages/firebase_ml_vision/android/src/main/java/io/flutter/plugins/firebasemlvision/FirebaseMlVisionPlugin.java]: Generally we avoid printing in the plugins. Just pass the exception up and let it be handled by the caller.

File: packages/firebase_ml_vision/android/src/main/java/io/flutter/plugins/firebasemlvision/FirebaseMlVisionPlugin.java [hunk 4]
```diff
@@ -1,26 +1,123 @@
...
+      result.error("TextDetector#detectInImage", exception.getLocalizedMessage(), null);
```
Comment (kroikie) [inline packages/firebase_ml_vision/android/src/main/java/io/flutter/plugins/firebasemlvision/FirebaseMlVisionPlugin.java]: Could we use an error code that is different from the method being called. Eg: `textDetectorError`

File: packages/firebase_ml_vision/android/src/main/java/io/flutter/plugins/firebasemlvision/FirebaseMlVisionPlugin.java [hunk 5]
```diff
@@ -1,26 +1,123 @@
...
+                  blocks.add(blockData);
+                }
+                result.success(blocks);
+              }
+            })
+        .addOnFailureListener(
+            new OnFailureListener() {
+              @Override
+              public void onFailure(@NonNull Exception e) {
+                result.error("TextDetector#detectInImage", e.getLocalizedMessage(), null);
```
Comment (kroikie) [inline packages/firebase_ml_vision/android/src/main/java/io/flutter/plugins/firebasemlvision/FirebaseMlVisionPlugin.java]: Use a more descriptive error code.

File: packages/firebase_ml_vision/example/lib/main.dart [hunk 6]
```diff
@@ -2,32 +2,154 @@
...
+
+  Future<void> _getImage() async {
+    setState(() {
+      _imageFile = null;
+      _imageSize = null;
+      _textLocations = null;
+    });
+
+    final File imageFile =
+        await ImagePicker.pickImage(source: ImageSource.camera);
```
Comment (kroikie) [inline packages/firebase_ml_vision/example/lib/main.dart]: Could we use ImageSource.gallery so when using an emulator the sample is just as useful.

**Pull request merged**